### ISSUE TITLE
Scoreboard improvements

### ISSUE THREAD

**Issue by pandaro** [yvt/openspades#301]
In the list of players (the one that appears when you press TAB) should also appear vital statistics of a player (for example "ratio").
Next to each player, you should see the icon of the weapon in use (SMG, rifle, shotgun).

**Comment by pandaro** [comment]
this issues can be close
@yvt

**Comment by pandaro** [comment]
Sorry do not close this issue cause i want to try to change the scoreboard by my self.
The main idea is:
-make the scoreboard completely autoscalable based on screen resolution
-and at second point, make it much more informative.
for now the info that i can take are:
name,
number,
score,
live or dead,
weapon type,
color.
What do you think about the chance to see all this information about your teammates? and on your opponents?

**Comment by VelocityRa** [comment]
You still working on this? Or can we close this?

**Comment by pandaro** [comment]
i have other project now, but i was in a good status. make what do you  want. if you like i can share code

**Comment by sanchez-k** [comment]
Hi guys. I like, how one user, named DD, have  made a nice OS Mod, and how he realised weapon info at scoreboard. Just look, that we see every player weapon. Here is a link [http://aloha.pk/index.php?topic=8558.0](url) for his topic. 

He left his code changes:
> I have attached the modified source code for all files that I've changed. All changes have been kept track of using // ADDED: and // MODIFIED: comments. Also in Sources.rar is a text file containing info about any Resources that have been changed.

Hope, this can be useful for future release.



![weapon_info](https://cloud.githubusercontent.com/assets/6381978/20769232/b88e6376-b749-11e6-948f-b55d5bf1530e.png)
[Sources.zip](https://github.com/yvt/openspades/files/622889/Sources.zip)

**Comment by VelocityRa** [comment]
@sanchez-k Thank you, I wasn't aware of this. I don't think _implementation_ is the problem however.
Like YVT said above, being able to know other player's weapons might be considered cheating as it cannot be done in the vanilla client.
So, I think this issue needs the NG label.

**Comment by pandaro** [comment]
ok, the question about the cheating is the same that i  have solved.
The solution is check the team and show info only about your team

**Comment by VelocityRa** [comment]
Hm, I suppose showing only your own team's weapons is ok.

**Comment by feikname** [comment]
I am confused, is this next-gen or not?

**Comment by pandaro** [comment]
i think a new protocol is not needed, just a nice new layout

**Comment by VelocityRa** [comment]
Not sure tbh, displaying your own team's weapons sounds acceptable (probably), but the enemies' is definitely not if players from other clients can't see them (which they're not).

**Comment by VelocityRa** [comment]
Also, most of the stuff you're suggesting (the description/title for this Issue is pretty vague) sound like next-gen protocol stuff (based on what yvt said), so I'm leaving the label there. A new Issue can be opened about (specifically) weapon icons for the player's team, or other features which might be possible (and fair) on the current server protocol.

**Comment by sanchez-k** [comment]
**VelocityRa** Just want to comment one thing about cheating aspect. 
For example - on aloha.pk, the most popular server in AoS and OS community, you can find common stats about every server. [http://stats.aloha.pk:1337/](url) 
There is described a lot of stats - and weapons in particular, including enemies. For ex. Babel - [Babel ststs on aloha.pk](http://199.127.56.82:34886/) So, I can simply go there and watch everyting i need. 
I don't think, that info about weapon selection will give me any advantage over enemies IMHO.
![os stats](https://cloud.githubusercontent.com/assets/6381978/20798310/17f113d4-b7e7-11e6-9684-232c363fe740.png)
Maybe this or other improvement will make our game more interactive.

Other way, that not all servers provide this stats info. But I hope, that you will make  fair decision in future game development.
Thank you for all your work and next release!

**Comment by VelocityRa** [comment]
@sanchez-k Thank you as well! I wasn't aware they exposed all this real-time information. I don't think we should make decisions like that based on one server though (albeit a very popular one). Exposing this information isn't that "cheaty" to begin with, however...
Personally I don't want it strongly enough to do it myself, so it's up to YVT or theunknownxy either way, at least for now.

**Comment by feikname** [comment]
I think an middle-groind is to make the feature in a hackish way, displaying the weapons for both teams if the joined server contains any special tag (`OS_displayAllPlayerWeapons`) but it sounds too hackish for me and I doubt any server will change.

**Comment by Kurtoid** [comment]
I think only showing weapons for the players team (or all for a spectator)
is a good idea.

**Comment by feikname** [comment]
I guess it's okay for same team and spectators at least.

**Comment by pandaro** [comment]
i think i nice info for other team player can be:
-how many ammo and grenade a player have.
- ratio and other skills can be useful in order to vote kick and to know if a player can be a good partner in a raid

**Comment by VelocityRa** [comment]
@pandaro Yeah but we're only talking next-gen here, neither of these are possible with the current server protocol unfortunately.
